### ISSUE TITLE
[docs] Make HCL table headers more discoverable

### ISSUE THREAD

**Issue by ninavizz** [QubesOS/qubes-issues#6784]
### The problem you're addressing (if any)

@andrewdavidwong recently added the terrific functionality to the HCL table, of sortable headers. Hooray! Unfortunately, a few small interactive and visual cues are missing, to let folks know that in fact, these headers are sortable.

**Today's HCL Table**

No cursor-change or color change on hover, to signify to users the item is live.

https://user-images.githubusercontent.com/8262612/126083911-36ed2bdc-75f2-46e8-ba12-6cabe5c0a1ac.mov

### Describe the solution you'd like

Generally I'm a strong advocate of interaction patterns documented by [patternfly.org](https://www.patternfly.org/v4) (maintained by RedHat, oddly enough!) or [Material.io](https://material.io/components/data-tables#anatomy) as a solid reference—yet both have specs for sortable headers, that kinda stink. Menus are more common to sort tables with, these days, as a pattern with richer options, than just table headers—which I suspect is why. 

_Key attributes that are missing from today's implementation are:_
- The user's cursor should change when hovering over the general area of a sortable header
- The color of the header text should change on hover
- An icon of some sort should be visible both on static and on hover, to indicate sortability
- When the list changes direction, the arrow(s) should indicate that. 

**Wikipedia**
Wikipedia's sortable table was mentioned in the Forum. It is highly usable, and is identical to the Bootstrap 4 spec for sortable table headers, but it is also quite busy/cluttered and lacks bolded headers that change color on hover. Like, the arrows don't **_have_** to be that big... but I'm also sure this implementation gets an A+ accessibility rating, which is not easy.

https://user-images.githubusercontent.com/8262612/126083921-d396f730-b3cf-474d-b7f7-cc9b9cd19783.mov

**GDrive**
Two different types of sorting. I do like the dot behind the arrow, as an extra "ting" of discoverability.
[This is the Material Style for Tables](https://material.io/components/data-tables/web#data-tables), that might be an option to implement?

https://user-images.githubusercontent.com/8262612/126083901-84c7c452-9ecc-4ec2-85c7-4ae6714c80d1.mov

**My Bank**
Note that often in tables, some headers are sortable and others are not. Users will probably expect that in this table, too. While I feel the arrows here and in the Wikipedia example, above, are needlessly clunky—you can't deny that they're not clear. 

https://user-images.githubusercontent.com/8262612/126083896-c85ef6ff-31cd-4a5c-88d4-d8acfab203e0.mov

**Harvest**
My favorites of the ones I just poked around for, is from my invoicing software (Harvest). My only nits with it, are that 1. There should be pale gray arrows on hover, and 2. The column widths forcing the arrow to wrap onto the next line, is weird. 

https://user-images.githubusercontent.com/8262612/126083906-04f41124-3894-4c1b-840a-b89c29af9a38.mov

TL;DR, my fav is the Google Material table, first. If you could literally implement just that. The Bootstrap Table standard would also get the job done. :)

### Where is the value to a user, and who might that user be?

Discussed [in forums](https://forum.qubes-os.org/t/plans-to-add-search-filter-functionality-to-the-hcl/5099).

### Describe any alternative solutions you've considered

The current first implementation of sorting is a great first step, but regrettably the sorting is not discoverable.

### Additional context

Nopitty nope, nope.

-----

**Comment by andrewdavidwong** [comment]
> Wikipedia's sortable table was mentioned in the Forum. It is highly usable, and is identical to the Bootstrap 4 spec for sortable table headers...

> The Bootstrap Table standard would also get the job done. :)

We're actually already using Bootstrap, but I'm not sure if it's Bootstrap 4. Are you saying that this is already included and that there's an easy way to apply it?

**Comment by ninavizz** [comment]
@andrewdavidwong `mdbootstrap` and `bootstraptable` are two projects that build components off of Bootstrap, that I'd found the examples from. My bad, I just combed the Bootstrap 4 page and found nothing directly from the Bootstrap project that is plug-n-play. :(

https://examples.bootstrap-table.com/

https://mdbootstrap.com/docs/b4/jquery/tables/sort/

**Comment by Tie-fighter** [comment]
Not sure if this should be a separate ticket, but the sorting is also acting in unexpected ways:

![Screenshot](https://user-images.githubusercontent.com/170315/126336813-4ff670d8-8160-4933-abde-4bd745c4d3cb.png)

Maybe a wrong sorting modus (numerical instead of lexical)?

**Comment by andrewdavidwong** [comment]
> Not sure if this should be a separate ticket, but the sorting is also acting in unexpected ways:
> 
> 
> 
> ![Screenshot](https://user-images.githubusercontent.com/170315/126336813-4ff670d8-8160-4933-abde-4bd745c4d3cb.png)
> 
> 
> 
> Maybe a wrong sorting modus (numerical instead of lexical)?

Yes, separate issue for this, please.

**Comment by ninavizz** [comment]
Actually, a quick hack that could help would just be inserting a line of text between the H2 and the table, saying "Tip: Click on headers to sort." Random thought as I'm using the page, myself, right now.

**Comment by andrewdavidwong** [comment]
> Actually, a quick hack that could help would just be inserting a line of text between the H2 and the table, saying "Tip: Click on headers to sort." Random thought as I'm using the page, myself, right now.

Done: https://github.com/QubesOS/qubesos.github.io/commit/39e9d8b8755d0636af5104255a2b54a244dbd846
